python/TestHarness/testers/Tester.py

```diff
@@ -71,21 +71,52 @@ def __init__(self, name, params):
 
     # Set the status message
     if self.specs['check_input']:
-      self.status_message = 'SYNTAX PASS'
+      self.success_message = 'SYNTAX PASS'
     else:
-      self.status_message = self.specs['status_message']
+      self.success_message = self.specs['success_message']
+
+    # Initialize the status bucket
+    self.status = {}
+
+    # Initialize the 'reason' a test failed. Empty means it passed.
+    self.status_message = ''
 
     # Set up common paramaters
     self.should_execute = self.specs['should_execute']
     self.check_input = self.specs['check_input']
+    self.success_bucket = 'PASS'
 
   # Method to return the input file if applicable to this Tester
   def getInputFile(self):
     return None
 
+  # Method to return the successful message printed to stdout
+  def getSuccessMessage(self):
+    return self.success_message
+
+  # Method to return status text (exodiff, crash, skipped because x, y and z etc)
   def getStatusMessage(self):
     return self.status_message
 
+  # Method to return enumerated status (PASS, FAIL, SKIP, DIFF)
+  def getStatus(self):
+    if self.getStatusMessage() in self.status:
+      return self.status[self.getStatusMessage()]
+    return None
+
+  # Method to set the bucket status
+  def setStatus(self, reason, bucket):
+    self.status_message = reason
+    self.status[reason] = bucket
+    return self.getStatus()
+
+  # Method to check for successfull test
+  def didPASS(self):
+    if self.getStatusMessage() == '':
+      self.setStatus(self.getSuccessMessage(), self.success_bucket)
+      return True
+    return False
```

Comment:
A couple of things about this routine:
1) Capitalization is wonky, let's just use `didPass`
2) This method mutates the state of the object. That's not what I'd expect from a method called `didPass`. What if the status is already set but not necessarily something that's bad (like "LAUNCHED"), does that mean that this method would return failed?